mesh/mesh.go

```diff
@@ -389,6 +359,52 @@ func (vl *validator) ValidateLayer(ctx context.Context, lyr *types.Layer) {
 	logger.Info("done validating layer")
 }
 
+func (msh *Mesh) persistLayerHashes(ctx context.Context, from, to types.LayerID) error {
+	if to.Before(from) {
+		msh.WithContext(ctx).With().Panic("verified layer went backward",
+			log.FieldNamed("fromLayer", from),
+			log.FieldNamed("toLayer", to))
+	}
+
+	msh.WithContext(ctx).With().Debug("persisting layer hashes",
+		log.FieldNamed("from_layer", from),
+		log.FieldNamed("to_layer", to))
+	for i := from; !i.After(to); i = i.Add(1) {
```

Comment:
i suggest to initialize logger once for this method and use it, there is a certain overhead associated with log.WithContext. see https://github.com/spacemeshos/go-spacemesh/issues/2657

Reply:
done